componentLibraries/defaultLibrary/Special/AeroComponents/AeroFlightRecorder.hpp

```diff
@@ -0,0 +1,329 @@
+
+ Copyright 2017 Hopsan Group
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+
+ The full license is available in the file LICENSE.
+ For details about the 'Hopsan Group' or information about Authors and
+ Contributors see the HOPSANGROUP and AUTHORS files that are located in
+ the Hopsan source code root directory.
+
+-----------------------------------------------------------------------------*/
+
+//!
+//! @file   AeroFlightRecorder.hpp
+//! @author Alejandro Sobron <alejandro.sobron@liu.se>
+//! @date   2020-05-15
+//!
+//! @brief Contains a log-file generator component for geographical visualization
+//! @ingroup AeroComponents
+//!
+
+#ifndef AEROFLIGHTRECORDER_HPP_INCLUDED
+#define AEROFLIGHTRECORDER_HPP_INCLUDED
+
+#include "ComponentEssentials.h"
+#include "ComponentSystem.h"
+#include "ComponentUtilities.h"
+#include "math.h"
+#include <iostream>
+#include <fstream>
+#include <ctime>
+
+namespace hopsan {
+
+    class AeroFlightRecorder : public ComponentSignal
+    {
+
+    private:
+    // Private member variables
+
+        //inputVariables pointers
+        double *mpLongitude;
+        double *mpLatitude;
+        double *mpAltitude;
+        double *mpPhi;
+        double *mpTheta;
+        double *mpPsi;
+        double *mpV;
+
+        //inputParameters pointers
+        double *mpDt;
+        double *mpScaleX;
+        double *mpScaleY;
+        double *mpScaleZ;
+
+        //inputVariables
+        double longitude;
+        double latitude;
+        double altitude;
+        double roll;
+        double pitch;
+        double heading;
+        double speed;
+
+        //inputParameters
+        double dt;
+        double scaleX;
+        double scaleY;
+        double scaleZ;
+
+        // Constants
+        enum FileTypes {FlightPathKML, GroundPathKML, ThreeDmodelKML};
+        int mFileType;
+        HString mFilePath;
+        HString mDaeModelPath;
+        std::ofstream mFile;
+
+        // Other operating variables
+        double tilt;
+        double mLastLogTime;
+
+    public:
+    // The creator function that is registered when a component lib is loaded into Hopsan
+        static Component *Creator()
+        {
+            return new AeroFlightRecorder();
+        }
+
+        void configure()
+        {
+            //Add ports to the component
+
+            //Add inputVariables to the component
+            addInputVariable("longitude","Longitude E","deg",15.577,&mpLongitude);
```

Comment:
I realize lat,lon is easiest to work with in deg, but it would maybe better to use the Quantity Angle. But right now it is not possible to choose the display unit for the quantity to use by default so it would then show up in rad in the GUI. But it would be convertible to deg.
I am not sure what to do here right now, because if we go with deg, we wont be able to change it later as "old models" would then have the values expressed in degrees instead of the Angle base unit.

What do you think @robbr48, maybe we should use Angle here,  and later add default unit selection to the caf xml file, so that the component creator can choose the default scaling to be visible. 

If you change to Angle, then you should write deg2rad(15.577) and deg2rad(58.401) in this initialization for readability.
Those functions are likely available through the ComponentUtilities.h header (actually AuxiliarySimulationFunctions.h)

Reply:
Yeah, this is a tricky one. I thought about restricting the input type to degrees since it's kind of standard, and this component was meant to work together with the EarthCoordinateSystem component (whose output is in deg, even though the component shows "m"(!) ).

But I think what you suggest is more reasonable: using the generic "Angle", making it clear in the documentation, and later if possible preselecting the default unit to "deg". I think I'd go for this if no better idea is proposed.

I wasn't aware that such a function (deg2rad) was available. Absolutely! I'll use it then.

Reply:
Yes EarthCoordinateSystem seems to have the registered unit incorrect. While we can change the C++ code this is a generated component, so the compgen code should be fixed as well. Maybe @petkr14 can help with this.

The problem here is that if the output is in degrees, we cant just connect it to an Angle input. Because the data in the model will be degrees and Angle expects rad. Any scaling or changing units is only for plotting purpose int he GUI, it does not affect the simulation data.
Of course you could put a converter gain in between.

So in this case degrees would  make sense, but then it wont be easy to change to Quantity Angle in the future. Because then old models will become incorrect. HopsanCore can not check if the "unit" is correct, a signal is only data, and the unit/quantity is only a text string.
I guess some checks could be done, at least to see if Quantity matches, but we are not doing that right now.

But maybe degrees makes sense for lat, lon. Thats is how its used most often.
Maybe lat lon would deserve its own "fictive quantity", but I dont think that is a good idea.

For now I guess you must choose, either degrees and then it will stay degrees, which is reasonable in this case. Or go with Angle but then there will be a mismatch with some other "Aerocomponents"